src/test/java/com/neo4j/docker/neo4jserver/TestBasic.java

```diff
@@ -202,11 +205,48 @@ void testShutsDownCleanly(String signal) throws Exception
     @Test
     void testStartsWhenDebuggingEnabled()
     {
-        try(GenericContainer container = createBasicContainer())
+        try ( GenericContainer container = createBasicContainer() )
         {
             container.withEnv( "NEO4J_DEBUG", "true" );
             container.start();
             Assertions.assertTrue( container.isRunning() );
         }
     }
+
+    @Test
+    void testContainerIsHealthyWhenNeo4jIsListeningAtPort7474()
+    {
+        try ( var container = createBasicContainer() )
+        {
+            container.setWaitStrategy( Wait.forHealthcheck() );
+            container.start();
+
+            Assertions.assertTrue( container.isRunning() );
+            Assertions.assertEquals( "healthy", container.getCurrentContainerInfo().getState().getHealth().getStatus() );
+        }
+    }
+
+    @Test
+    void testContainerIsHealthyWhenListenAddressIsModifiedByUser()
+    {
+        assertContainerWithListenAddressIsHealthy( ":4747" );
+        assertContainerWithListenAddressIsHealthy( "127.0.0.1:7474" );
+        assertContainerWithListenAddressIsHealthy( "localhost:7474" );
+        assertContainerWithListenAddressIsHealthy( "localhost" );
+    }
+
+    void assertContainerWithListenAddressIsHealthy( String listenAddress )
```

Comment:
To make this test parameterised with JUnitParams, replace this line with:
```
@ParameterizedTest
@ValueSource(strings = {":4747", "127.0.0.1:7474", "localhost:7474", "localhost"})
void testContainerIsHealthyWhenListenAddressIsModifiedByUser(String listenAddress)
```
and delete the old `testContainerIsHealthyWhenListenAddressIsModifiedByUser` test.

For further reference see:
https://github.com/Pragmatists/JUnitParams/blob/master/src/test/java/junitparams/usage/SamplesOfUsageTest.java

Reply:
Noice